src/low-level/smtp/mailsmtp.c

```diff
@@ -530,6 +530,9 @@ int mailsmtp_data_message_quit_no_disconnect(mailsmtp * session,
     return MAILSMTP_ERROR_STREAM;
   
   r = send_quit(session);
+  if (r != MAILSMTP_NO_ERROR)
+    return MAILSMTP_ERROR_STREAM;
```

Comment:
`return r` would be more appropriate here right?

Reply:
Done.